scim/resource_service_principal.go

```diff
@@ -161,7 +182,27 @@ func ResourceServicePrincipal() *schema.Resource {
 			})
 		},
 		Delete: func(ctx context.Context, d *schema.ResourceData, c *common.DatabricksClient) error {
-			return NewServicePrincipalsAPI(ctx, c).Delete(d.Id())
+			spAPI := NewServicePrincipalsAPI(ctx, c)
+			if c.AccountID != "" {
+				return nil
+			}
+			var err = spAPI.Delete(d.Id())
+			if err != nil {
+				return err
+			}
+			if d.Get("delete_repos").(bool) {
+				err = spAPI.DeleteRepos(d.Id())
+				if err != nil {
+					return err
+				}
+			}
+			if d.Get("delete_home_dir").(bool) {
+				err = spAPI.DeleteHomeDirectory(d.Id())
```

Comment:
please add `acceptance.Test` (the one talking to the real workspace) to verify that homedirs are not deleted if this flag is not set and another test to verify that home dir is deleted, when it's set